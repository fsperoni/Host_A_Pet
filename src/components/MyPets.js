import React, { useContext, useEffect, useState } from 'react';
import UserContext from "./UserContext";
import AddPetForm from './AddPetForm';
import HostAPetApi from './Api';

const MyPets = () => {
  const { currentUser } = useContext(UserContext);
  const [pets, setPets] = useState([]);

  useEffect(function getPets() {

    const getUserPets = async() => {
      const userPets = await HostAPetApi.getPets(currentUser.username);
      setPets(userPets);
    }
    getUserPets();
  }, [currentUser.username]);

  const petCards = pets.map(pet => (
    <div key={pet.id} className="card">
      <div className="card-body">
        <p>{pet.name} the {pet.type}</p>
        <p>{pet.photo}</p>
      </div>
    </div>
  ))
  return (
    <div className="MyPets">
      <div className="container col-md-6 offset-md-3 col-lg-4 offset-lg-4">
        <h2 className="mb-3">My Pets</h2>
        {petCards}
      </div>
      <AddPetForm pets={pets} setPets={setPets}/>
    </div>
  )
}

export default MyPets;