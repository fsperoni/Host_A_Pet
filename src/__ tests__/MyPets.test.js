import { render } from "@testing-library/react";
import UserContext from '../hooks/useUserContext';
import MyPets from "../components/MyPets";
import { BrowserRouter } from "react-router-dom";

const {
  currentAdminUser,
} = require("./_testCommon");

// smoke test
it('renders without crashing', function () {
  render(
    <BrowserRouter>
      <UserContext.Provider
        value={{ currentUser: currentAdminUser(), function() { } }}>
        <MyPets />
      </UserContext.Provider>
    </BrowserRouter>
  );

});

//snapshot test
it("matches snapshot", function () {
  const { asFragment } = render(
    <BrowserRouter>
      <UserContext.Provider
        value={{ currentUser: currentAdminUser(), function() { } }}>
        <MyPets />
      </UserContext.Provider>
    </BrowserRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("should have 'No pets added yet' in the component.", function () {
  const { getByText } = render(
    <BrowserRouter>
      <UserContext.Provider
        value={{ currentUser: currentAdminUser(), function() { } }}>
        <MyPets />
      </UserContext.Provider>
    </BrowserRouter>
  );
  expect(getByText("No pets added yet")).toBeInTheDocument();
  expect(getByText("My Pets")).toBeInTheDocument();
});
