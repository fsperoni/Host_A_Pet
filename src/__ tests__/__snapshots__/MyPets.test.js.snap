// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MyPets"
  >
    <div
      class="container col-md-6 offset-md-3 col-lg-4 offset-lg-4"
    >
      <h2
        class="mb-3"
      >
        My Pets
      </h2>
      <h3
        class="ms-2"
      >
        No pets added yet
      </h3>
    </div>
    <div
      class="AddPetForm mt-3 mb-3"
    >
      <div
        class="container col-md-6 offset-md-3 col-lg-4 offset-lg-4"
      >
        <h2
          class="mb-3"
        >
          Add a Pet
        </h2>
        <div
          class="card"
        >
          <div
            class="card-body"
          >
            <form>
              <div
                class="form-group"
              >
                <label>
                  Name:
                </label>
                <input
                  class="form-control"
                  name="name"
                  required=""
                  value=""
                />
              </div>
              <div
                class="form-group"
              >
                <label>
                  Type:
                </label>
                <select
                  class="form-select"
                  name="type"
                >
                  <option
                    required=""
                    value=""
                  >
                    Select one
                  </option>
                  <option
                    value="Cat"
                  >
                    Cat
                  </option>
                  <option
                    value="Dog"
                  >
                    Dog
                  </option>
                </select>
              </div>
              <div
                class="form-group"
              >
                <label>
                  Photo:
                </label>
                <input
                  class="form-control"
                  name="photo"
                  type="url"
                  value=""
                />
              </div>
              <div
                class="container mt-4"
              >
                <button
                  class="btn btn-primary btn-block"
                >
                  Add
                </button>
                <button
                  class="btn btn-warning btn-block ms-4"
                >
                  Cancel
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
